### ISSUE TITLE
Create an Android client

### ISSUE THREAD

**Issue by rsertelon** [JCDecaux/opendata-client#2]
This client library works well in JEE apps, as it is based upon Jax-RS and RESTeasy.

I was thinking about potential Android developers who might want to include our data into their apps. It would be nice if they could use this library too.

To do so, it would require us to change the maven project a bit.

I was thinking of:

```
opendata-client-parent
  -- opendata-client-jaxrs (RESTeasy impl)
  -- opendata-client-android (Retrofit impl)
```

Might do this tonight.

**Comment by rsertelon** [comment]
Retrofit documentation: http://square.github.io/retrofit/#restadapter-configuration

**Comment by rsertelon** [comment]
Retrofit client is in master. However, it's not been tested yet.

I would use some advice. @JoanZapata @NicolasPoirier @synapticvoid do you think the implementation is okay? Should I use HttpClient, or ship OkHttp?

Thanks for your feedback :)

**Comment by JoanZapata** [comment]
Looks good. I don't think you should ship OkHttp since Retrofit will use it "if present" at runtime.

**Comment by rsertelon** [comment]
Yep, but I need to set proxy settings, and in doing this, I have no choice, I can either:

* Choose a default client, this is what I'm doing no with Apache HttpClient
* Re-code the same logic as in Retrofit for choosing the client depending on classpath
* Try to contribute to retrofit so they can have Proxy support

IMHO, the third solution would be the best. But I have to check out how I can do that.

**Comment by JoanZapata** [comment]
I see...

What about using the Android built-in http client by default but add a constructor overload that would take a retrofit client as a parameter?

**Comment by rsertelon** [comment]
I would rather abstract this for the users. Because de config is the same on every platform then. (UX does not change).

**Comment by JoanZapata** [comment]
Then I would go for 2.
